### PR TITLE
test covers the new feature Alert improvements Follow-ups

### PR DESCRIPTION
**Pull request by DanielOsypenko**
Test OCS-4844 steps:

1. clone ODF monitoring tool from [https://url.corp.redhat.com/odf-monitoring-tools ](https://url.corp.redhat.com/odf-monitoring-tools%C2%A0) 
2. clone upstream and downstream versions of prometheus rules files
3. run 'comparealerts' against downstream and upstream prometheus rules YAML files
4. run 'comparealerts' against two identical prometheus rules YAML files, check no "No diffs found"
5. replace alert name in one of identical prometheus rules YAML file, run 'comparealerts' and verify deviations found
6. convert two prometheus rules files with deviation to JSON and verify deviations found
7. run 'comparealerts' against two identical prometheus rules JSON files, check "No diffs found"

### CODE REVIEW

**Comment by fbalak** [comment]
What is odf-monitoring-tools? There is no documentation or docstrings in the project.

File: tests/internal/test_monitoring_tool.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            prometheus_alerts_upstream
+        ), f"cannot find '{prometheus_alerts_upstream}' in cloned repo"
+        logger.debug(f"print '{prometheus_ocs_rules}' file")
+        exec_cmd(f"cat {prometheus_ocs_rules}")
+        logger.debug(f"print '{prometheus_alerts_upstream}' file")
+        exec_cmd(f"cat {prometheus_alerts_upstream}")
+
+        logger.info("compare upstream and downstream prometheus rule files")
+        complete_proc = exec_cmd(
+            f"./comparealerts {prometheus_ocs_rules} {prometheus_alerts_upstream}"
```
Comment (fbalak) [inline tests/internal/test_monitoring_tool.py:150]: https://github.com/red-hat-storage/ocs-operator.git is not a downstream repo. This test doesn't take into account version of installed cluster, only what is in master of the repo.

File: tests/internal/test_monitoring_tool.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    assert all(
+        alert in compare_res for alert in deviation_list
+    ), f"compare tool did not find all occurancies from {deviation_list}:\n{compare_res}"
+
+
+@tier1
+@blue_squad
+@polarion_id("OCS-4844")
+class TestMonitoringTool(BaseTest):
+    def test_odf_monitoring_tool(
```
Comment (fbalak) [inline tests/internal/test_monitoring_tool.py:118]: Please add a docstring with description of the test.

**Review by fbalak — CHANGES_REQUESTED** [review]

File: tests/internal/test_monitoring_tool.py (hunk 1 above)
Comment (DanielOsypenko) [reply tests/internal/test_monitoring_tool.py:150]: this test doesn't require cluster, and any version of the OCP, ODF deployment accordingly. 
relatively one to another and between two files:
prometheus_ocs_rules - came from our 'downstream' repo https://github.com/red-hat-storage/ocs-operator.git
prometheus_alerts_upstream - came from the 'upstream' ceph repo https://github.com/ceph/ceph.git

that is what I understood from conversation with devel assignee

**Comment by DanielOsypenko** [comment]
> What is odf-monitoring-tools? There is no documentation or docstrings in the project.

added explanation to the test

**Review by fbalak — COMMENTED** [review]
Please split the code to multiple files: fixtures to conftest.py (you can create a new one in tests/internal/) and functions related to test to it's module (maybe ocs_ci/utility/monitoring_tool.py?).

**Comment by DanielOsypenko** [comment]
> ocs_ci/utility/monitoring_tool.py

@fbalak 
done. job ran locally -> https://drive.google.com/file/d/1c4K-_X7ePDM7BqIpb5Yfy6oxSJebwEqg/view?usp=share_link

**Review by fbalak — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Pull request merged**